dev/generate_test_trajectory.ts

```diff
@@ -1,144 +1,450 @@
+/**
+ * Exports trajectory with movement data between work and home,
+ * but no clusters at either of these locations.
+ */
+function createMobileOnlyTrajectory(testBase: TrajectoryTestBase): Trajectory {
+  const trajectoryHomeToWork = addTimestampsForTrajectory(
+    trajectoryTimes.homeEndDate,
+    trajectoryTimes.workStartDate,
+    testBase.homeToWorkTrajectory
+  )
+
+  const trajectoryWorkToHome = addTimestampsForTrajectory(
+    trajectoryTimes.workEndDate,
+    trajectoryTimes.homeAfterWorkStartDate,
+    testBase.workToHomeTrajectory
+  )
+  const trajectoryMobileOnly = combineTrajectories(
+    {
+      id: testDataNames.mobileOnly,
+      placename: testDataNames.mobileOnly,
+      type: TrajectoryType.EXAMPLE,
+    },
+    [trajectoryHomeToWork, trajectoryWorkToHome]
+  )
+  return trajectoryMobileOnly
 }
 
+/**
+ * Exports trajectory with movement data between work and home.
+ * Contains few  basic locations at home & work, but no clusters.
+ */
+function createHomeWorkTrajectory(testBase: TrajectoryTestBase): Trajectory {
+  const trajectoryHome = addTimestampsForTrajectory(
+    trajectoryTimes.homeStartDate,
+    trajectoryTimes.homeEndDate,
+    testBase.homeTrajectory
+  )
+  const trajectoryHomeToWork = addTimestampsForTrajectory(
+    trajectoryTimes.homeEndDate,
+    trajectoryTimes.workStartDate,
+    testBase.homeToWorkTrajectory
+  )
+  const trajectoryWork = addTimestampsForTrajectory(
+    trajectoryTimes.workStartDate,
+    trajectoryTimes.workEndDate,
+    testBase.workTrajectory
+  )
+  const trajectoryWorkToHome = addTimestampsForTrajectory(
+    trajectoryTimes.workEndDate,
+    trajectoryTimes.homeAfterWorkStartDate,
+    testBase.workToHomeTrajectory
+  )
+  const trajectoryHomeAfterWork = addTimestampsForTrajectory(
+    trajectoryTimes.homeAfterWorkStartDate,
+    trajectoryTimes.homeAfterWorkEndDate,
+    testBase.homeTrajectory
+  )
+  const trajectoryHomeWork = combineTrajectories(
+    {
+      id: testDataNames.homeWork,
+      placename: testDataNames.homeWork,
+      type: TrajectoryType.EXAMPLE,
+    },
+    [
+      trajectoryHome,
+      trajectoryHomeToWork,
+      trajectoryWork,
+      trajectoryWorkToHome,
+      trajectoryHomeAfterWork,
+    ]
+  )
+  return trajectoryHomeWork
+}
+
+/**
+ * Exports trajectory with movement data between work and home,
+ * which includes temporally sparse clusters at both locations.
+ * Contains roughly one location per hour per cluster.
+ */
+function createTemporallySparseTrajectory(
+  testBase: TrajectoryTestBase
+): Trajectory {
+  const trajectoryHomeTemporallySparse = addTimestampsForTrajectory(
+    trajectoryTimes.homeStartDate,
+    trajectoryTimes.homeEndDate,
+    createCluster(
+      testBase.homeTrajectory,
+      Math.round(
+        getTimeDiffInHours(
+          trajectoryTimes.homeStartDate,
+          trajectoryTimes.homeEndDate
+        )
+      )
+    )
+  )
+  const trajectoryHomeToWorkTemporallySparse = addTimestampsForTrajectory(
```

Comment:
Since the commutes are actually not sparse / we do not change them to make them sparse, sth. like `trajectoryHomeToWork` could be sufficient?